iac-extensions/arm/src/main/java/org/sonar/iac/arm/parser/bicep/BicepGrammar.java

```diff
@@ -89,11 +94,65 @@ public VariableDeclaration VARIABLE_DECLARATION() {
         b.token(EOL)));
   }
 
+  public ResourceDeclaration RESOURCE_DECLARATION() {
+    return b.<ResourceDeclaration>nonterminal(BicepLexicalGrammar.RESOURCE_DECLARATION).is(
+      f.resourceDeclaration(
+        b.token(BicepKeyword.RESOURCE),
+        IDENTIFIER(),
+        INTERPOLATED_STRING_TYPE(),
+        b.optional(
+          f.ignoreFirst(
+            b.token(BicepLexicalGrammar.SPACING),
+            b.token(BicepKeyword.EXISTING))),
+        b.token(Punctuator.EQU),
+        OBJECT_EXPRESSION(),
+        b.token(BicepLexicalGrammar.EOL)));
+  }
+
+  public InterpolatedString INTERPOLATED_STRING_TYPE() {
+    return b.<InterpolatedString>nonterminal(BicepLexicalGrammar.INTERPOLATED_STRING).is(
+      f.interpolatedString(
+        b.optional(b.token(BicepLexicalGrammar.SPACING)),
+        b.token(BicepLexicalGrammar.APOSTROPHE),
+        b.token(BicepLexicalGrammar.QUOTED_STRING_LITERAL),
+        b.token(BicepLexicalGrammar.APOSTROPHE)));
+  }
+
+  // object -> "{" ( NL+ ( property NL+ )* )? "}"
+  public ObjectExpression OBJECT_EXPRESSION() {
+    return b.<ObjectExpression>nonterminal(BicepLexicalGrammar.OBJECT_EXPRESSION).is(
+      f.objectExpression(
+        b.token(Punctuator.LCURLYBRACE),
+        b.optional(PROPERTY_LIST()),
+        b.token(Punctuator.RCURLYBRACE)));
+  }
+
+  public List<Property> PROPERTY_LIST() {
+    return b.<List<Property>>nonterminal().is(
+      f.propertyList(
+        f.ignoreFirst(
+          b.oneOrMore(
+            b.token(BicepLexicalGrammar.EOL)),
+          b.zeroOrMore(
+            f.ignoreSecond(
+              PROPERTY(),
+              b.token(BicepLexicalGrammar.EOL))))));
```

Comment:
This is much too complex and not following the reference grammar. The identifier of the property should be prefixed by a SPACE which includes optional line terminators. Please see the PHP [implementation](https://github.com/SonarSource/sonar-php/blob/89b09df773f71eec3c14b47731d309a78e5f0fe5/php-frontend/src/main/java/org/sonar/php/parser/PHPLexicalGrammar.java#L314)